test/NuGet.Core.Tests/NuGet.DependencyResolver.Core.Tests/RemoteDependencyWalkerTests.cs

```diff
@@ -1840,6 +1840,44 @@ public async Task WalkAsync_WithDowngradeInPrunedSubgraph_DoesNotReportDowngrade
             result.Downgrades.Should().BeEmpty();
         }
 
+        /// <summary>
+        /// A 1.0 -> D 1.0 (Central transitive)
+        ///       -> B 1.0 -> D 3.0 (Central transitive - should be ignored)
+        ///                -> C 1.0 -> D 2.0
+        /// </summary>
+        [Fact]
+        public async Task TransitiveDependenciesFromNonRootLibraries_AreIgnored()
+        {
+            var context = new TestRemoteWalkContext();
+            var provider = new DependencyProvider();
+            provider.Package("A", "1.0")
+                .DependsOn("B", "1.0")
+                .DependsOn("D", "1.0", LibraryDependencyTarget.Package, versionCentrallyManaged: true, libraryDependencyReferenceType: LibraryDependencyReferenceType.None);
```

Comment:
Do we another another level of dependencies here?

If A is the root node, and it's centrally managed, then all it's dependencies need to be centrally managed and are direct.